src/rgw/rgw_rest_s3.cc

```diff
@@ -6224,18 +6224,33 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
     dump_errno(s);
   }
 
+  auto bl_len = [](bufferlist& bl){int i=0;for(auto& iter : bl.buffers())i++; return i;};
+
   // Explicitly use chunked transfer encoding so that we can stream the result
   // to the user without having to wait for the full length of it.
   if (chunk_number == 0) {
     end_header(s, this, "application/xml", CHUNKED_TRANSFER_ENCODING);
   }
 
   int status=0;
+  int i=0;
+
   for(auto& it : bl.buffers()) {
+
+    ldout(s->cct, 10) << "processing segment " << i << "out of " << bl_len(bl) << " off " << ofs
+                      << " len " << len << " obj-size " << s->obj_size << dendl;
+
+    if(it.length() == 0)
+    {
```

Comment:
please format according to the ceph [CodingStyle](https://github.com/ceph/ceph/blob/master/CodingStyle#L84-L103)
```c++
if (it.length() == 0) {
```